src/SqlStreamStore.Postgres/PostgresStreamStore.ReadAll.cs

```diff
@@ -18,68 +23,363 @@ partial class PostgresStreamStore
+                    Logger.ErrorFormat(
+                        "{correlation} Possible SKIPPED EVENT as we will stop waiting for in progress transactions! One of these transactions: {transactions} is in progress for longer then {totalTime}ms",
+                        correlation,
+                        transactionsInProgress,
+                        _settings.GapHandlingSettings.PossibleAllowedSkipTime);
+                    return;
+                }
+
+                if(totalTime > _settings.GapHandlingSettings.PossibleDeadlockTime)
+                {
+                    Logger.ErrorFormat(
+                        "{correlation} Possible DEADLOCK! One of these transactions: {transactions} is in progress for longer then {totalTime}ms",
+                        correlation,
+                        transactionsInProgress,
+                        _settings.GapHandlingSettings.PossibleDeadlockTime);
+                }
+
+
+                if(delayTime > 0)
+                {
+                    Logger.TraceFormat("{correlation} Delay 'PollTransactions' for {delayTime}ms",
+                        correlation,
+                        delayTime);
+                    await Task.Delay(delayTime, cancellationToken);
+                }
+
+                if(count % 5 == 0)
+                    delayTime += 10;
+
+                totalTime += sw.ElapsedMilliseconds;
+                count++;
+                sw.Restart();
+
+                Logger.TraceFormat(
+                    "{correlation} State 'PollTransactions' | count: {count} | delayTime: {delayTime} | totalTime: {totalTime}",
+                    correlation,
+                    count,
+                    delayTime,
+                    totalTime);
+            } while(stillInProgress);
+        }
+
+        private async Task<(List<(StreamMessage message, int? maxAge)>, bool)> ReadTrustedMessages(
+            long fromPositionInclusive,
+            long toPositionInclusive,
+            int maxCount,
+            bool prefetch,
+            Guid correlation,
+            CancellationToken cancellationToken)
+        {
+            Logger.TraceFormat("{correlation} Read trusted message initiated", correlation);
+
+            var (readAllResult, _) = await ReadAllForwards(fromPositionInclusive,
+                maxCount,
+                prefetch,
+                correlation,
+                cancellationToken);
+
+            if(!readAllResult.Any())
+            {
+                Logger.TraceFormat("{correlation} Read trusted message result is empty.", correlation);
+                return (new List<(StreamMessage message, int? maxAge)>(), true);
+            }
+
+            Logger.TraceFormat(
+                "{correlation} Filter messages from {fromPositionInclusive} to {toPositionInclusive}",
+                correlation,
+                fromPositionInclusive,
+                toPositionInclusive);
+
+            var readResultToReturn = readAllResult.Where(x =>
```

Comment:
Wouldn't it be nicer if we actually only retrieve those messages instead of a LINQ query in memory.
This LINQ query can maybe take a while if you retrieve a lot of messages. (which we do for our resyncer)